### PR TITLE
Account for removal of `analysis_options_user.yaml` from Flutter

### PR DESCRIPTION
**Pull request by parlough**
This file no longer exists after https://github.com/flutter/flutter/pull/138261 and is generally no longer supported.

Also update nearby code with expanded comments and avoiding the non-null assertions.

### CODE REVIEW

**Review by isoos — COMMENTED** [review]
Needs an update for the golden files:
- `rm test/goldens/end2end/*json*`
- `dart test test/end2end_test.dart`

**Comment by isoos** [comment]
I think you may have deleted all the files and not just `*json*`, which needs a full re-run if you change SDKs (as the dependencies will change).

**Comment by parlough** [comment]
Thanks for the help with the goldens! I think I was only considering my Dart version and needed to use a matching Flutter build.

File: lib/src/analysis_options.dart [hunk 1]
```diff
@@ -8,92 +8,53 @@ import 'dart:isolate';
...
+/// The default analysis options configuration (in its raw yaml format).
+Future<String> getDefaultAnalysisOptionsYaml() async =>
+    await _getLintsCoreAnalysisOptions();
 
 Future<String> _getLintsCoreAnalysisOptions() async {
-  // try to load local lints from the resolved package references
-  if (_cachedLintsCoreInResolvedReferences != null) {
-    return _cachedLintsCoreInResolvedReferences!;
+  // Try to load local lints from the resolved package references.
+  if (_cachedLintsCoreInResolvedReferences case final cachedCoreLints?) {
```
Comment (isoos) [inline lib/src/analysis_options.dart:25]: I don't find this more readable than the `!= null` version. Could you please change it back (also the one below).

**Review by isoos — CHANGES_REQUESTED** [review]

File: lib/src/analysis_options.dart (hunk 1 above)
Comment (parlough) [reply lib/src/analysis_options.dart]: Fair! The syntax does take a bit of getting used to :)

I reverted this change but then ended up simplifying the implementation a bit further as we ended up only using one of the cached fields anyway. So this private method is now agnostic of the caching. Let me know what you think, thanks!

**Review by isoos — APPROVED** [review]

**Pull request merged**